pkg/blobstore/metrics_blob_access.go

```diff
@@ -127,20 +150,31 @@ func (ba *metricsBlobAccess) FindMissing(ctx context.Context, digests digest.Set
 	timeStart := ba.clock.Now()
 	digests, err := ba.blobAccess.FindMissing(ctx, digests)
 	ba.updateDurationSeconds(ba.findMissingDurationSeconds, status.Code(err), timeStart)
+	span.SetStatus(errToStatus(err))
 	return digests, err
 }
 
 type metricsErrorHandler struct {
 	blobAccess *metricsBlobAccess
 	timeStart  time.Time
 	errorCode  codes.Code
+	span       *trace.Span
 }
 
 func (eh *metricsErrorHandler) OnError(err error) (buffer.Buffer, error) {
 	eh.errorCode = status.Code(err)
+	eh.span.SetStatus(errToStatus(err))
 	return nil, err
 }
 
 func (eh *metricsErrorHandler) Done() {
 	eh.blobAccess.updateDurationSeconds(eh.blobAccess.getDurationSeconds, eh.errorCode, eh.timeStart)
+	eh.span.End()
+}
+
+func errToStatus(err error) trace.Status {
+	if err == nil {
+		return trace.Status{}
```

Comment:
I've seen us juggle with trace.Status in a couple of other places. Should we add a proper utility function for this to pkg/util/status.go?

Reply:
I was suspecting the same thing, let me work this bit into the opentelemetry conversion PR.